agent/functional_tests/util/utils.go

```diff
@@ -220,6 +221,48 @@ func (agent *TestAgent) StartTaskWithTaskDefinitionOverrides(t *testing.T, task
 	return tasks[0], nil
 }
 
+// StartAWSVPCTask starts a task with "awsvpc" networking mode
+func (agent *TestAgent) StartAWSVPCTask(t *testing.T, task string) (*TestTask, error) {
+	td, err := GetTaskDefinition(task)
+	if err != nil {
+		return nil, err
+	}
+
+	return agent.startAWSVPCTask(t, td)
+}
+
+func (agent *TestAgent) startAWSVPCTask(t *testing.T, taskDefinition string) (*TestTask, error) {
+	t.Logf("Task definition: %s", taskDefinition)
+	// Get the subnet ID, which is a required parameter for starting
+	// tasks in 'awsvpc' network mode
+	subnet, err := getSubnetID()
+	if err != nil {
+		return nil, err
+	}
+
+	t.Logf("Starting 'awsvpc' task in subnet: %s", subnet)
+	resp, err := ECS.StartTask(&ecs.StartTaskInput{
+		Cluster:            &agent.Cluster,
+		ContainerInstances: []*string{&agent.ContainerInstanceArn},
+		TaskDefinition:     &taskDefinition,
+		NetworkConfiguration: &ecs.NetworkConfiguration{
+			AwsvpcConfiguration: &ecs.AwsVpcConfiguration{
+				Subnets: []*string{&subnet},
+			},
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+	if len(resp.Failures) != 0 || len(resp.Tasks) == 0 {
+		return nil, errors.New("Failure starting task: " + *resp.Failures[0].Reason)
+	}
+
+	task := resp.Tasks[0]
+	agent.t.Logf("Started task: %s\n", *task.TaskArn)
```

Comment:
Why `agent.t` instead of `t`?

Reply:
because it's there :) I'll modify this so as to not pass the `testing.T` object again